### DISCUSSION TITLE
AsyncExitStackMiddleware middleware.starlette takes a lot of time

### DISCUSSION THREAD

**Discussion by theobouwman** [encode/starlette#2188] [Potential Issue] [answered]
I am monitoring my API with Sentry and noted that AsyncExitStackMiddleware takes a lot of time... Is this normal?

<img width="1182" alt="Screenshot 2023-06-16 at 17 29 03" src="https://github.com/encode/starlette/assets/16098190/3e339e81-dde6-4c28-ba11-0dadc37e5ffd">

**Comment by Kludex** [comment, marked as answer]
That middleware is from FastAPI. 👀 

Not sure why it says `starlette` on Sentry... 🤷‍♂️

**Reply by theobouwman** [reply]
I will ask ik there.

**Comment by adriangb** [comment]
It's the teardown for your dependencies, e.g. closing/committing a database session.
